crates/daemon/src/monitor.rs

```diff
@@ -478,10 +481,14 @@ impl Actor {
 
         tracing::trace!("Updating status of {num_transactions} transactions",);
 
-        let histories = self
-            .client
-            .batch_script_get_history(self.state.monitoring_scripts())
-            .context("Failed to get script histories")?;
+        let scripts = self.state.monitoring_scripts().cloned().collect();
+
+        let mut rx_script_history = batch_script_get_history(self.client.clone(), scripts);
+
+        let mut histories = Vec::new();
+        while let Some(script_history) = rx_script_history.recv().await {
+            histories.push(script_history)
+        }
```

Comment:
yeah, this is pretty much what I was doing last night!

some things for consideration:
- I would have put that inside the `batch_script_get_history` to have it encapsulated (otherwise, we're pointlessly leaking the internal implementation with the receiver (we can just expose the stream if we don't want to collect to the vector)

Reply:
I did this in https://github.com/itchysats/itchysats/pull/3197/commits/f8532aeff0a7a1a316a8b03110a9539d36ef387d
Given the context of this PR it makes sense to do this.

Note that the granularity and assertions of the tests changed. 
On the long run I would push for #3198 which will require processing on the outside, but we can discuss these changes on #3198